### PR TITLE
Update core dependency to google-cloud-core >= 0.23.0, < 0.24dev.

### PR DESCRIPTION
**Pull request by daspecster**
Updates 'google-cloud-core >= 0.23.0, < 0.24dev' dependencies in services.

### CODE REVIEW

**Comment by daspecster** [comment]
Premature? This can wait if needed.

File: vision/setup.py [hunk 1]
```diff
@@ -51,7 +51,7 @@
 
 REQUIREMENTS = [
     'enum34',
-    'google-cloud-core >= 0.22.1, < 0.23dev',
+    'google-cloud-core >= 0.23., < 0.24dev',
```
Comment (lukesneeringer) [inline vision/setup.py]: Typo.

**Comment by daspecster** [comment]
@lukesneeringer I haven't checked the clients. I feel like that should be a separate PR?

**Comment by lukesneeringer** [comment]
@dhermes said (in an instant message) that it is already done.

**Comment by lukesneeringer** [comment]
@dhermes Why is merging this premature? It seems reasonable to me.

**Comment by dhermes** [comment]
Feel free to merge. No worries.

**Pull request merged**